### DISCUSSION TITLE
Coding-Question

### DISCUSSION THREAD

**Discussion by ghost** [SkriptLang/Skript#7198] [General]
Hello, I looked over the last changes and noticed, that in Variable.java the following has been changed:
`setIndex(event, "" + i + SEPARATOR + j, ((Object[]) value)[j]);`
-> 
`setIndex(event, "" + i + SEPARATOR + (j + 1), ((Object[]) value)[j]);`

Any ideas why this got changed?

**Comment by sovdeeth** [comment]
this was changed back in 2.8.0, as part of #5878, to fix bugs with ExprParse.
Nothing recent.

**Comment by ghost** [comment]
I see - sorry that I ask but what exactly is this setIndex(event, "" + i + SEPARATOR + (j + 1), ((Object[]) value)[j]); for?
I thought it changes the default index from 0 to 1 in list, but it does not do that.

**Comment by ghost** [comment]
i dont even get this case triggered.

**Reply by sovdeeth** [reply]
I'm not really sure what's got you so focused on this. It just handles cases that only pop up with ExprParse parsing lists.

**Reply by ghost** [reply]
the syntax is still incorrect.

**Reply by sovdeeth** [reply]
Huh? It works as intended, it's not incorrect.

**Reply by ghost** [reply]
Could you share an example code?

**Reply by sovdeeth** [reply]
```set {_list::*} to "5, 4, 3, 2, and 1 and stone sword" parsed as "%numbers% and %item%"```

**Reply by ghost** [reply]
how would I access it then? {_list::1::1} would be 5, right?
